dev/bin/kafka-setup.sh

```diff
@@ -0,0 +1,123 @@
+  fi
+
+  mc_sa_name="${MANAGED_KAFKA_INSTANCE_NAME}-mc"
+  create_service_account "${mc_sa_name}"
+  mc_sa_credentials_file="${CREDENTIALS_FOLDER}/${mc_sa_name}.json"
+  mc_sa_id=$( jq -r '.clientID' "${mc_sa_credentials_file}" )
+  if [ "${sa_updated}" == "yes" ]; then
+    rhoas kafka acl grant-admin -y --service-account "${mc_sa_id}"
+    rhoas kafka acl create -y --user "${mc_sa_id}" --permission allow --operation all --group all
+    rhoas kafka acl create -y --user "${mc_sa_id}" --permission allow --operation all --topic all
+    rhoas kafka acl create -y --user "${mc_sa_id}" --permission allow --operation all --transactional-id all
+    echo "Managed Connector account: ACLs created"
+  fi
+}
+
+function create_service_account {
+  sa_name="$1"
+  sa_count=$( rhoas service-account list -o json | jq -rc ".items[] | select( .name == \"${sa_name}\" )" | wc -l )
+  sa_credentials_file="${CREDENTIALS_FOLDER}/${sa_name}.json"
+  sa_updated="no"
+
+  if [ $sa_count -gt 1 ]; then
+    die "ERROR: there are ${sa_count} service accounts named \"${sa_name}\""
+  elif [ $sa_count -eq 0 ]; then
+    echo "Creating service account named \"${sa_name}\"..."
+    rhoas service-account create --output-file="${sa_credentials_file}" --file-format=json --overwrite --short-description="${sa_name}"
+    echo "Created service account named \"${sa_name}\""
+    sa_updated="yes"
+  else
+    echo "Service account named \"${sa_name}\" found"
+  fi
+
+  sa_id=$( rhoas service-account list -o json | jq -rc ".items[] | select( .name == \"${sa_name}\" ) | .id" )
+  if ! [ -f "${sa_credentials_file}" ]; then
+    echo "No credentials file found for service account named \"${sa_name}\". Resetting credentials..."
+    rhoas service-account reset-credentials --id "${sa_id}" --output-file="${sa_credentials_file}" --file-format=json -y
+    sa_updated="yes"
+  fi
+}
+
+# create kafka instance and wait for it to be ready
+function create_kafka_instance_and_wait_ready {
+  # create instance if not already existing
+  instance_count=$( rhoas kafka list --search "${MANAGED_KAFKA_INSTANCE_NAME}" -o json | jq -rc ".items[] | select( .name == \"${MANAGED_KAFKA_INSTANCE_NAME}\" )" | wc -l )
+  if [ $instance_count -gt 1 ]; then
+    die "ERROR: there are ${instance_count} instances named \"${MANAGED_KAFKA_INSTANCE_NAME}\""
+  elif [ $instance_count -eq 0 ]; then
+    echo "Creating Managed Kafka instance named \"${MANAGED_KAFKA_INSTANCE_NAME}\"..."
+    rhoas kafka create --name "${MANAGED_KAFKA_INSTANCE_NAME}"
+    echo "Created Managed Kafka instance named \"${MANAGED_KAFKA_INSTANCE_NAME}\""
+  else
+    echo "Managed Kafka instance named \"${MANAGED_KAFKA_INSTANCE_NAME}\" found"
+  fi
+
+  # set instance as current
+  instance_id=$( rhoas kafka describe --name "${MANAGED_KAFKA_INSTANCE_NAME}" -o json | jq -rc '.id' )
+  rhoas kafka use --id "${instance_id}"
+
+  # wait for instance to be ready
+  kafka_status=$( rhoas status kafka -o json | jq -rc '.kafka.status' )
```

Comment:
there may be several Kafka instances, what about filtering the proper instance first:
`select( .kafka.name == ${MANAGED_KAFKA_INSTANCE_NAME} ) | .kafka.status`

Reply:
As far as I know `rhoas status kafka` returns the status of the current kafka cluster in use, so it will always be one (which in our case is configured in the comment above). Am I missing something @sutaakar ?

Reply:
looks that for some reason the `rhoas status kafka` doesn't return any info even when I setup current Kafka using `rhoas kafka use --id <some id>`
not sure why, may be a bug. Using rhoas version 0.38.1.

Reply:
I have a previous version (0.36.6). Let me check.

Reply:
@sutaakar I have your same error with rhoas v0.38.2. It seems a bug. However we can achieve the same behavior with `rhoas kafka describe`.

Reply:
@sutaakar I updated the script, please take a look and let me know

Reply:
The command seems passing now, will try whole script

Reply:
@kostola / @sutaakar Can one of you please report the bug for `v.0.38.2` to the DevExp team (or check that its expected behaviour). Thanks.

Reply:
Reported as https://github.com/redhat-developer/app-services-cli/issues/1425

Reply:
should be fixed in next release